init.d/net.lo.in

```diff
@@ -118,6 +118,31 @@ _flatten_array()
 	_array_helper $1
 }
 
+_wait_for_presence()
+{
+	local timeout=
+
+	_exists && return 0
+
+	eval timeout=\$presence_timeout_${IFVAR}
+	timeout=${timeout:-${presence_timeout:-0}}
+
+	[ ${timeout} -le 0 ] && return 1
+
+	einfon "Waiting for ${IFACE} to show up (${timeout} seconds)"
+	while [ ${timeout} -gt 0 ]; do
+		_exists && break
+
+		sleep 1
```

Comment:
please look at the other sleep loop code that supports sub-second resolution for sleeps. 1 second is sometimes overkill.

Reply:
This function is very similar to "_wait_for_carrier" which also has 1 second resolution.
What kind of resolution would you want this to accept instead? Deciseconds?